libmscore/keysig.cpp

```diff
@@ -101,7 +101,8 @@ void KeySig::layout()
             }
 
       _sig.keySymbols().clear();
-      if (staff() && !staff()->genKeySig())     // no key sigs on TAB staves
+      // no key sigs on TAB staves or if the staff type says no key sigs
+      if (staff() && (!staff()->genKeySig() || !staff()->staffType(tick())->genKeysig()))
```

Comment:
You might consider just adding the tick parameter to Staff::genKeySig(), that's how most other tick-dependent staff properties are handled.  Llook at, for example, Staff::small()

Reply:
Ah okay that makes sense, I will look into that! Thank you!

Reply:
Also, there shouldn't be a need to test both the global setting and the tick-based setting.  We don't do that elsewhere.  The tick-based check should do the right thing no matter what.

Reply:
Yeah I will take out the first part of the OR. They do the same thing except the first part checks from the beginning of the score and the other one checks from the current tick which is what should be happening. Thank you.